tests/integration/crunchers/test_cruncher_time_dep_quantile_rolling_windows.py

```diff
@@ -0,0 +1,165 @@
+        error_msg = re.escape(
+            "Not all required times in the dictionary have data in the database."
+        )
+        times_quant = {2011: 0.5}
+        with pytest.raises(ValueError, match=error_msg):
+            tcruncher.derive_relationship(
+                "Emissions|CO2", ["Emissions|CH4"], times_quant
+            )
+
+    def test_derive_relationship_wrong_times_input(self, test_db):
+        tcruncher = self.tclass(test_db)
+        # Due to a current bug in pyam, we need to ensure this is not a 64-type value
+        t_0 = list(test_db[test_db.time_col].unique())[0]
+        times_quant = {t_0: 0.5}
+        res = tcruncher.derive_relationship(
+            "Emissions|CO2", ["Emissions|CH4"], times_quant
+        )
+        error_msg = re.escape(
+            "Not all required times in the infillee database can be found in "
+            "the dictionary."
+        )
+        if test_db.time_col == "year":
+            filtered_db = test_db.filter(year=int(t_0), keep=False)
+        else:
+            filtered_db = test_db.filter(time=self._convert_dt64_todt(t_0), keep=False)
+        with pytest.raises(ValueError, match=error_msg):
+            res(filtered_db)
+
+    @pytest.mark.parametrize("timecol", ["year", "time"])
+    def test_relationship_usage(self, timecol):
+        # Define a regular set of values and check that we return the correct quantiles
+        # of it
+        if timecol == "year":
+            dates = [2010, 2020, 2030]
+        else:
+            dates = [
+                datetime(year=2010, month=6, day=6),
+                datetime(year=2020, month=6, day=6),
+                datetime(year=2030, month=6, day=6),
+            ]
+        regular_db = pd.DataFrame(
+            [
+                [_ma, _sa + str(val), "World", _eco2, _gtc, val, val, val]
+                for val in range(11)
+            ],
+            columns=_msrvu + dates,
+        )
+        regular_data = IamDataFrame(regular_db)
+        regular_data["variable"] = _ech4
+        regular_data["value"] = 1
+        regular_data.append(IamDataFrame(regular_db), inplace=True)
+        regular_data = IamDataFrame(regular_data.data)
+        tcruncher = self.tclass(regular_data)
+        follow = _eco2
+        formatted_dates = list(regular_data[regular_data.time_col].unique())
+        quant = {
+            formatted_dates[0]: 0.4,
+            formatted_dates[1]: 0.5,
+            formatted_dates[2]: 0.6,
+        }
+        res = tcruncher.derive_relationship(
+            follow, [_ech4], time_quantile_dict=quant, nwindows=1,
+        )
+        to_infill = regular_data.filter(variable=_ech4)
+        returned = res(to_infill)
+        for time, quantile in quant.items():
+            if timecol == "year":
+                filtered_ans = returned.filter(year=int(time))["value"]
+            else:
+                filtered_ans = returned.filter(time=self._convert_dt64_todt(time))[
```

Comment:
You can import `_convert_dt64_todt` from `silicone.database_crunchers.time_dep_quantile_rolling_windows` instead of defining it as a method of this test class.

Reply:
It takes more writing but I guess it's cleaner